### PR TITLE
feat(fedora): release upgrade script

### PR DESCRIPTION
**Pull request by Angxddeep**
## Type of Change
- [x] New feature

## Description
An easy way to upgrade to the next version of fedora, not recommended (by me) for gnome and KDE spin because they support upgrades through their software store.

## Testing
I have tested everything except the upgrade itself, I am currently on fedora 41 and in no mood of reinstalling my install, but the code is good so it will work.

## Additional Information
I couldn't understand what task_list =  means in tab data, so If someone could explain what it is, I would really appreciate it
## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by jeevithakannan2** [comment]
This was not implemented as it could break systems. Make sure you've tested it.

**Comment by jeevithakannan2** [comment]
Try testing it in a VM.

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+# Global version variables
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:5]:
```suggestion
```

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+current_version=$(rpm -E '%{fedora}')
+next_version=$((current_version + 1))
+previous_version=$((current_version - 1))
+
+# Function to update Fedora
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:10]:
```suggestion
```

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            "$ESCALATION_TOOL" "$PACKAGER" system-upgrade reboot
+        else
+            printf "%b\n" "${YELLOW}You can reboot later to apply the upgrade.${RC}"
+        fi
+    else
+        printf "%b\n" "${RED}No upgrade performed.${RC}"
+    fi
+}
+
+# Function to run post-upgrade tasks
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:49]:
```suggestion
```

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            "$ESCALATION_TOOL" "$PACKAGER" distro-sync -y
+            ;;
+        *)
+            printf "%b\n" "${RED}Unsupported package manager: $PACKAGER.${RC}"
+            exit 1
+            ;;
+    esac
+}
+
+# Main menu
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:65]:
```suggestion
```

File: core/tabs/system-setup/tab_data.toml [hunk 5]
```diff
@@ -59,6 +59,12 @@ description = "RPM Fusion provides software that the Fedora Project or Red Hat d
 script = "fedora/rpm-fusion-setup.sh"
 task_list = "MP"
 
+[[data.entries]]
+name = "Upgrade to a New Fedora Release"
+description = "Upgrades the System to the next fedora release"
```
Comment (adamperkowski) [inline core/tabs/system-setup/tab_data.toml:64]:
```suggestion
description = "Upgrades system to the next Fedora release"
```

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+. ../../common-script.sh
+
+# Global version variables
+current_version=$(rpm -E '%{fedora}')
+next_version=$((current_version + 1))
+previous_version=$((current_version - 1))
+
+# Function to update Fedora
+update() {
+    printf "%b\n" "${RED}Make sure your system is fully updated; if not, update first and reboot once.${RC}"
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:12]:
```suggestion
    printf "%b\n" "${RED}Make sure your system is fully updated; if not, update it first and reboot once.${RC}"
```

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    printf "%b\n" "${CYAN}Your current Fedora version is $current_version.${RC}"
+    printf "%b\n" "${CYAN}The next available version is $next_version.${RC}"
+    
+    printf "%b\n" "${YELLOW}Do you want to update to Fedora version $next_version? (y/n): ${RC}"
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:16]:
```suggestion
    printf "%b\n" "${YELLOW}Do you want to update to $next_version? (y/n): ${RC}"  
```

**Comment by adamperkowski** [comment]
Yeah I agree with @jeevithakannan2

**Comment by adamperkowski** [comment]
Yeah I agree with @jeevithakannan2

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 8]
```diff
@@ -0,0 +1,85 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+# Global version variables
+current_version=$(rpm -E '%{fedora}')
+next_version=$((current_version + 1))
+previous_version=$((current_version - 1))
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:8]: it's unused

```suggestion
```

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+        if [ "$reboot_response" =~ ^[Yy]$ ]; then
+            printf "%b\n" "${GREEN}Rebooting to apply the upgrade...${RC}"
+            "$ESCALATION_TOOL" "$PACKAGER" system-upgrade reboot
+        else
+            printf "%b\n" "${YELLOW}You can reboot later to apply the upgrade.${RC}"
+        fi
+    else
+        printf "%b\n" "${RED}No upgrade performed.${RC}"
+    fi
```
Comment (adamperkowski) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:46]: bashisms

```suggestion
    case "$response" in
        y|Y)
            printf "%b\n" "${CYAN}Preparing to update to $next_version...${RC}"
        
            if ! "$ESCALATION_TOOL" "$PACKAGER" install dnf-plugin-system-upgrade -y; then
                printf "%b\n" "${RED}Failed to install dnf-plugin-system-upgrade.${RC}"
                exit 1
            fi
        
            if ! "$ESCALATION_TOOL" "$PACKAGER" system-upgrade download --releasever="$next_version"; then
                printf "%b\n" "${RED}Failed to download the upgrade packages.${RC}"
                exit 1
            fi

            printf "%b\n" "${YELLOW}Do you want to reboot now to apply the upgrade? (y/n): ${RC}"
            read -r reboot_response

            case "$reboot_response" in
                y|Y)
                    printf "%b\n" "${GREEN}Rebooting to apply the upgrade...${RC}"
                    "$ESCALATION_TOOL" "$PACKAGER" system-upgrade reboot
                    ;;
                *)
                    printf "%b\n" "${YELLOW}You can reboot later to apply the upgrade.${RC}"
                    ;;
            esac
            ;;
        *)
            printf "%b\n" "${RED}No upgrade performed.${RC}"
            ;;
    esac
}
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 10]
```diff
@@ -0,0 +1,83 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+current_version=$(rpm -E '%{fedora}')
+next_version=$((current_version + 1))
+
+update() {
+    printf "%b\n" "${RED}Make sure your system is fully updated; if not, update it first and reboot once.${RC}"
```
Comment (jeevithakannan2) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:9]:
```suggestion
    printf "%b\n" "${YELLOW}Make sure your system is fully updated; if not, update it first and reboot once.${RC}"
```

File: core/tabs/system-setup/fedora/fedora-upgrade.sh [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+                printf "%b\n" "${RED}Failed to download the upgrade packages.${RC}"
+                exit 1
+            fi
+
+            printf "%b\n" "${YELLOW}Do you want to reboot now to apply the upgrade? (y/n): ${RC}"
+            read -r reboot_response
+
+            case "$reboot_response" in
+                y|Y)
+                    printf "%b\n" "${GREEN}Rebooting to apply the upgrade...${RC}"
```
Comment (jeevithakannan2) [inline core/tabs/system-setup/fedora/fedora-upgrade.sh:35]:
```suggestion
                    printf "%b\n" "${YELLOW}Rebooting to apply the upgrade...${RC}"
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

**Review by adamperkowski — APPROVED** [review]
> Just tested it on real hardware and its working

Well if you say so.

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**